pkg/controller/machinehealthcheck/machinehealthcheck_controller_test.go

```diff
@@ -1827,14 +1855,14 @@ func TestNeedsRemediation(t *testing.T) {
 			},
 			timeoutForMachineToHaveNode: defaultNodeStartupTimeout,
 			expectedNeedsRemediation:    false,
-			expectedNextCheck:           time.Duration(1 * time.Minute), // 300-200 rounded
+			expectedNextCheck:           time.Minute, // 300-200 rounded
 			expectedError:               false,
 		},
 	}
 
 	for _, tc := range testCases {
 		t.Run(tc.testCase, func(t *testing.T) {
-			needsRemediation, nextCheck, err := tc.target.needsRemediation(tc.timeoutForMachineToHaveNode)
+			needsRemediation, nextCheck, err := tc.target.needsRemediation(tc.timeoutForMachineToHaveNode, 0)
```

Comment:
Should we not adjust this to cater for having some failed machines? This is currently just skipping over the logic that was added in this PR

Reply:
I don't think so.
Please note that- the code you are referring is part of `TestNeedsRemediation` method, where our tests are in `TestReconcile` method.
This change was required because of the change in the method signature.